src/main/java/com/devonfw/quarkus/productmanagement/service/v1/ProductRestService.java

```diff
@@ -127,14 +128,14 @@ public ProductDto getProductByTitle(@PathParam("title") String title) {
   }
 
   @APIResponses({
-  @APIResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = ProductDto.class))),
-  @APIResponse(responseCode = "404", description = "Product not found"), @APIResponse(responseCode = "500") })
+  @APIResponse(responseCode = "204", description = "OK", content = @Content(schema = @Schema(implementation = ProductDto.class))),
+  @APIResponse(responseCode = "500", description = "Product not found"), @APIResponse(responseCode = "500") })
```

Comment:
Not found should always be simply 404 to be more restful.
See my comments on JAX-RS exception support and use NotFoundException that is already there.
Also we discussed that annotating all this stuff to every REST method is completly redundant and therefore anti-pattern. We want to try servicedocgen that has meanwhile been improved.

Reply:
Also we have to clearly distinguish the following cases:
* a valid ID is given (4711) but no such entity is found in DB --> 404 NotFoundException
* an invalid ID ("doesnotexist") is provided --> 422 --> you can use your custom example exception for such case and even consider having a getStatusCode() method that returns Integer with null by default in abstract base exception (ApplicationException).
* an unexpected server error (e.g. connection to DB failed) --> 500 generic error message due to OWASP sensitive data exposure

It is important to properly distinguish 4xx and 5xx errors. Via our performance logging we collect all this data in log monitoring and via dashboards we need to distinguish what are errors caused by the client (3xx/4xx) and what are caused by the server indicating some operational problem and a potential outage (5xx).

Reply:
I agree that we should add the servicedocgen plugin and remove the comments here. I have created a following task for this (https://github.com/devonfw-sample/devon4quarkus-reference/issues/26)